include/deal.II/matrix_free/matrix_free.templates.h

```diff
@@ -373,6 +373,26 @@ namespace internal
                             (cell->level(), cell->index()));
       }
   }
+
+
+
+  // For serial Triangulations, we simply take all cells.
+  // In this case the subdomain_id does not correspond to the process id
+  // Hence, it is not necessarily constant.
+  template <typename InIterator>
+  void resolve_cell_serial (const InIterator   &cell,
+                            std::vector<std::pair<unsigned int,unsigned int> > &cell_its)
+  {
+    if (cell->has_children())
+      for (unsigned int child=0; child<cell->n_children(); ++child)
+        resolve_cell_serial (cell->child(child), cell_its);
+    else
+      {
+        Assert (cell->active(), ExcInternalError());
+        cell_its.push_back (std::pair<unsigned int,unsigned int>
+                            (cell->level(), cell->index()));
+      }
+  }
```

Comment:
The problem is not that `subdomain_id` is `numbers::invalid_subdomain_id` but that `subdomain_id` can be set to something arbitrary in user code. We do this for example ourselves in `dofs/dof_tools_common.h`.
Do I misunderstand what you are suggesting?